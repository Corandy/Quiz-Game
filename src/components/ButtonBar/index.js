import React from 'react';
import { Button } from 'react-bootstrap';

const ButtonBar = ({
        currentPage = 1, //number between min-max  
        min = 0, //number,
        max = false, //number or false 
        previousEvent = false, //event
        nextEvent = false, // event
        finishEvent = false // event
    }) => {
    return <div>
        {currentPage > min && <Button onClick={previousEvent} bsStyle="primary" type="submit">Previous</Button>}
        <span className={'page_indicator'}>{'Page '+currentPage}</span>
        {currentPage < max && <Button onClick={nextEvent} bsStyle="primary" type="submit">Next</Button> }
        {currentPage == max && !!finishEvent && <Button onClick={finishEvent} bsStyle="primary" type="submit">Finish</Button>}
    </div>;    
}

export default ButtonBar;