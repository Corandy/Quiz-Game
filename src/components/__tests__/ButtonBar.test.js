import React from 'react';
import { shallow } from 'enzyme';  
import ButtonBar from '../ButtonBar';

describe('ButtonBar', () => {

    it('renders without crashing', () => {
        shallow(<ButtonBar />);
    });

    it('renders correctly without max value', () => {
        const component = shallow(<ButtonBar currentPage={1} min={1}/>);
        expect(component).toMatchSnapshot();
    });

    it('renders correctly without Previous Button', () => {
        const component = shallow(<ButtonBar currentPage={1} min={1} max={10}/>);
        expect(component.find('.button_bar__previous').exists()).toEqual(false);
    });

    it('renders correctly without Next Button but with Finish Button', () => {
        const component = shallow(<ButtonBar currentPage={10} min={1} max={10} finishEvent={true}/>);
        expect(component.find('.button_bar__next').exists()).toEqual(false);
        expect(component.find('.button_bar__finish').exists()).toEqual(true);
    });

    const clickFn = jest.fn();
    it('Next Button is clickable', () => {
        const component = shallow(<ButtonBar currentPage={2} min={1} max={10} nextEvent={clickFn}/>);
        component
            .find('.button_bar__next')
            .simulate('click');
            expect(clickFn).toHaveBeenCalled();
    });

});