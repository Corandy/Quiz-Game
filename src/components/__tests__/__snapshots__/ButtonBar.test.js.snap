// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBar renders correctly without max value 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonBar
    currentPage={1}
    min={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <span
          className="page_indicator"
        >
          Page 1
        </span>,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Page 1",
          "className": "page_indicator",
        },
        "ref": null,
        "rendered": "Page 1",
        "type": "span",
      },
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <span
            className="page_indicator"
          >
            Page 1
          </span>,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page 1",
            "className": "page_indicator",
          },
          "ref": null,
          "rendered": "Page 1",
          "type": "span",
        },
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
